contracts/VotingMachines/AbsoluteVote.sol

```diff
@@ -95,11 +95,13 @@ contract AbsoluteVote is IntVoteInterface {
    * @param _proposalId the proposal ID
    */
     function cancelProposal(bytes32 _proposalId) external onlyProposalOwner(_proposalId) votable(_proposalId) returns(bool) {
+
         if (! parameters[proposals[_proposalId].paramsHash].allowOwner) {
             return false;
         }
+        address avatar = proposals[_proposalId].avatar;
```

Comment:
how about emit the event before deleting the proposal and save this assignment ?

Reply:
Are you sure? The pattern in all the other contracts is to emit *after* deletion.